src/libdredd/src/mutate_ast_consumer.cc

```diff
@@ -119,20 +121,32 @@ void MutateAstConsumer::HandleTranslationUnit(clang::ASTContext& ast_context) {
       // declaration. In such cases, no action is required.
       continue;
     }
-    auto source_range_in_main_file =
-        GetSourceRangeInMainFile(compiler_instance_->getPreprocessor(),
-                                 *constant_array_typeloc.getSizeExpr());
-    // We only consider the rewriting if the source range for the size
-    // expression is in the main source file.
-    if (source_range_in_main_file.isValid()) {
-      std::stringstream stringstream;
-      stringstream
-          << llvm::dyn_cast<clang::ConstantArrayType>(
-                 constant_sized_array_decl->getType()->getAsArrayTypeUnsafe())
-                 ->getSize()
-                 .getLimitedValue();
-      rewriter_.ReplaceText(source_range_in_main_file, stringstream.str());
-    }
+    RewriteExpressionInMainFileToIntegral(
+        constant_array_typeloc.getSizeExpr(),
+        llvm::dyn_cast<clang::ConstantArrayType>(
+            constant_sized_array_decl->getType()->getAsArrayTypeUnsafe())
+            ->getSize()
+            .getLimitedValue());
+  }
+
+  // Rewrite the argument of static assertion as needed to `1`.
+  // There's no need to evaluate the actual expression, as any value other than
+  // 1 would have caused the front-end used by Dredd to fail.
+  for (const auto& static_assert_decl :
+       visitor_->GetStaticAssertionsToRewrite()) {
+    RewriteExpressionInMainFileToIntegral(static_assert_decl->getAssertExpr(),
+                                          1);
+  }
+
+  // Rewrite the constant integer argument of some function.
```

Comment:
-> "Rewrite the constant integer arguments of some builtin functions."

(Is what I propose accurate?)

Reply:
Correct. I was considering the possibility of some other non-builtin functions that require constant integer argument, thus use more general naming(`constant_arguments_to_rewrite_.push_back` instead of  `constant_builtin_function_arguments_to_rewrite_`). 

For now, I will assume only "some builtin_function" need to have their argument rewritten. We can change the variable name and comment if we discover this is not the case in the future.